lab/lablib/midi/MidiFile.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * @enum {number}
+ */
+export const MIDIFormat = {
+  SINGLE_TRACK: 0,
+  MULTI_PARALLEL: 1,
+  MULTI_SEQUENCE: 2,
+};
+Object.freeze(MIDIFormat);
+
+export class MIDIHeader {
+  /**
+   * Constructor
+   * @param {MIDIFormat} format MIDI format
+   * @param {number} num_tracks Number of tracks in the file
+   * @param {number} ticks_per_quarter How many ticks per quarter note
+   */
+  constructor(format, num_tracks, ticks_per_quarter) {
+    this.format = format;
+    this.num_tracks = num_tracks;
+    this.ticks_per_quarter = ticks_per_quarter;
+  }
+}
+
+/**
+ * @enum {number}
+ */
+export const MIDIMessageType = {
+  NOTE_OFF: 0b100,
+  NOTE_ON: 0b1001,
+};
+Object.freeze(MIDIMessageType);
+
+export class MIDINote {
```

Comment:
Hm... when it comes to basic MIDI messages, they're always a status byte + 1-2 data bytes.
Note events are important and common enough to be a separate type, but for now the other message types can be ignored. 

Make a LUT of status byte to length, then parse it similarly to how it works for generic meta and sysex messages